bigtable/client/detail/readrowsparser.h

```diff
@@ -0,0 +1,132 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_READROWSPARSER_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_READROWSPARSER_H_
+
+#include "bigtable/client/version.h"
+
+#include "bigtable/client/cell.h"
+#include "bigtable/client/row.h"
+
+#include "google/bigtable/v2/bigtable.grpc.pb.h"
+
+#include <absl/strings/string_view.h>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * Transforms a stream of chunks as returned by the ReadRows streaming
+ * RPC into a sequence of rows.
+ *
+ * Users are expected to do something like:
+ *
+ * @code
+ * while (!stream.EOT()) {
```

Comment:
What is the plan in the case that `stream` fails with an error and the user wants to retry and continue?  Are the users supposed to create a new `ReadRowsParser`?  Or are they supposed to reset the parser somehow?

Either solution is fine, but you should document here how the class is used.

Reply:
I think a reasonable expectation is that a new parser is needed for each stream, and stream state is undefined if errors are ignored. Added as a comment.